File: App/module/vergleichsfunktionen.py
from helfer import *

# O(1)


def O(liste: list) -> int:
    return liste[0]

# O(log n)


def binärer_logorythmuen_von_n(liste: list) -> dict:
    return json_lesen(read(json_schreiben(function="Binärer_logorythmuen_von_n", liste=liste)))

# O(n)


def linar(liste: list) -> dict:
    return json_lesen(read(json_schreiben(function="Linar", liste=liste)))

# O(n log n)


def linear_mal_log_von_n(liste: list) -> dict:
    return json_lesen(read(json_schreiben(function="Linear_mal_log_von_n", liste=liste)))

# O(n^²)


def quadieren(liste: list) -> dict:
    return json_lesen(read(json_schreiben(function="Quadieren", liste=liste)))

# O(2^n)


def duplizieren(liste: list) -> dict:
    return json_lesen(read(json_schreiben(function="Duplizieren", liste=liste)))

File: App/module/vergleichsfunktionen.go
package main

import (
	"math"
)

// O(1)
func O(liste []int) int {
	return liste[0]
}

// O(log n)
func Binärer_logorythmuen_von_n(liste []int) []float64 {
	var ergebnisse []float64
	for _, x := range liste {
		ergebnisse = append(ergebnisse, math.Log2(float64(x)))
	}
	return ergebnisse
}

// O(n)
func Linar(liste []int) []int {
	var ergebnisse []int
	for _, x := range liste {
		ergebnisse = append(ergebnisse, x)
	}
	return ergebnisse
}

//O(n log n)

func Linear_mal_log_von_n(liste []int) []float64 {
	var ergebnisse []float64
	for _, x := range liste {
		ergebnisse = append(ergebnisse, float64(x)*math.Log2(float64(x)))
	}
	return ergebnisse
}

// O(n^²)
func Quadieren(liste []int) []float64 {
	var ergebnisse []float64
	for _, x := range liste {
		ergebnisse = append(ergebnisse, math.Pow(float64(x), 2.0))
	}
	return ergebnisse
}

// O(2^n)
func Duplizieren(liste []int) []float64 {
	var ergebnisse []float64
	for _, x := range liste {
		ergebnisse = append(ergebnisse, math.Pow(2.0, float64(x)))
	}
	return ergebnisse
}
